src/silverlabnwb/timings.py

```diff
@@ -0,0 +1,101 @@
+            self.n_lines_per_roi = n_y_pixels
+            self.n_pixels_per_line = n_x_pixels
+        else:
+            self.n_lines_per_roi = n_x_pixels
+            self.n_pixels_per_line = n_y_pixels
+        if self.n_pixels_per_line == 0 and self.n_lines_per_roi == 0:
+            # assume we are in pointing mode in this case, probably better if this were passed as an argument?
+            # this class would need to know about the Modes class then though.
+            self.n_pixels_per_line = 1
+            self.n_lines_per_roi = 1
+
+
+class LabViewTimingsPre2018(LabViewTimings):
+
+    def __init__(self, relative_times_path, roi_path, dwell_time):
+        super().__init__(relative_times_path, roi_path, dwell_time)
+        self._format_pixel_time_offsets()
+
+    def _read_relative_times_file(self, file_path):
+        raw_data = pd.read_csv(file_path, names=('RelativeTime', 'CycleTime'),
+                               sep='\t', dtype=np.float64) / 1e6  # convert to seconds
+        self.pixel_time_offsets = raw_data['RelativeTime']
+        self.cycle_time = raw_data['CycleTime'][0]
+
+    def _format_pixel_time_offsets(self):
+        row_increments = np.arange(self.n_pixels_per_line)*self.dwell_time
+        pixel_time_offsets_by_roi = {}
+        for roi_index in np.arange(self.n_rois):
+            start_index = self.n_lines_per_roi * roi_index
+            end_index = start_index + self.n_lines_per_roi
+            row_offsets = self.pixel_time_offsets[start_index:end_index]
+            pixel_time_offsets_by_roi[roi_index] = row_offsets[:, np.newaxis]+row_increments
+        self.pixel_time_offsets = pixel_time_offsets_by_roi
+
+
+class LabViewTimings231(LabViewTimings):
+
+    def __init__(self, relative_times_path, roi_path, dwell_time, n_cycles_per_trial, n_trials):
+        super().__init__(relative_times_path, roi_path, dwell_time)
+        self._format_pixel_time_offsets(n_cycles_per_trial, n_trials)
+
+    def _read_relative_times_file(self, file_path):
+        raw_data = pd.read_csv(file_path, sep='\t', dtype=np.float64) / 1e6
+        self.pixel_time_offsets = raw_data['Image Time [us]']
+        self.pixel_time_offsets = self.pixel_time_offsets[self.pixel_time_offsets != 0]
+
+    def _format_pixel_time_offsets(self, n_cycles_per_trial, n_trials):
+        pixel_time_offsets_by_roi = {}
+        n_lines_per_cycle = self.n_rois * self.n_lines_per_roi
+        row_increments = np.arange(self.n_pixels_per_line) * self.dwell_time
+        for roi_index in np.arange(0, self.n_rois):
+            pixel_time_offsets_by_roi[roi_index] = []
+            for cycle_index in np.arange(0, n_cycles_per_trial * n_trials):
+                start_index = self.n_lines_per_roi * roi_index + n_lines_per_cycle * cycle_index
+                end_index = start_index + self.n_lines_per_roi
+                pixel_time_offsets_by_roi[roi_index].append(self.pixel_time_offsets.values[start_index:end_index])
```

Comment:
It may be more efficient to preallocate an array of the required size, and proceed to fill it in for each cycle. It can probably be done without a loop, too. But this could also wait for another round of improvements.